src/components/App.vue

```
@@ -0,0 +1,60 @@
+<script>
+import Header from './Header.vue';
+import PostsTable from './PostsTable.vue';
+import Sidebar from './Sidebar.vue';
+import Loader from './Loader.vue';
+import Notification from './Notification.vue';
+
+export default {
+  components: { Header, PostsTable, Sidebar, Loader, Notification },
+  data() {
+    return {
+      posts: [],
+      selectedPost: null,
+      isCreating: false,
+      isLoading: false,
+      error: null,
+      user: null, // ✅ MUDAR DE HARD-CODED PARA NULL
+    };
+  },
+  methods: {
+    async fetchUser() {
+      try {
+        const res = await fetch('https://mate-academy.github.io/fe-students-api/users.json');
+        if (!res.ok) throw new Error('Failed to load user');
+        const users = await res.json();
+        this.user = users[0]; // ✅ USAR PRIMEIRO USUÁRIO DA API
+      } catch (error) {
+        console.error('Failed to load user:', error);
+        // Fallback para desenvolvimento
+        this.user = { id: 1, name: 'Fallback User' };
```

In the catch block you assign a hard-coded fallback user object (`this.user = { id: 1, name: 'Fallback User' }`). The task explicitly requires that the current user is obtained from the Users endpoint (not hard-coded). Using a fallback silently may violate that requirement and hide failures (tests may expect real userId). Prefer one of these approaches:
- Surface the error to the UI (set `this.error`) and keep `this.user` null so user-driven flows are disabled until a real user is obtained, or
- If you must provide a fallback for dev, gate it behind an explicit DEV flag and document it, and avoid using it for automated checks.

Also, if you choose to keep a fallback, make it explicit in UI/debug only and avoid silently creating posts with a hard-coded id.